File: demos/jans-chip/android/app/src/main/java/io/jans/chip/modal/JSONWebKeySet.java
package io.jans.chip.modal;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class JSONWebKeySet {

    public JSONWebKeySet(List<Map<String, ?>> keys) {
        this.keys = keys;
    }

    private List<Map<String, ?>> keys;

    public JSONWebKeySet() {
        keys = new ArrayList<>();
    }

    public List<Map<String, ?>> getKeys() {
        return keys;
    }

    public void setKeys(List<Map<String, ?>> keys) {
        this.keys = keys;
    }

    public void addKey(Map<String, ?> key){
        keys.add(key);
    }

    public String toJsonString() {
        ObjectMapper mapper = new ObjectMapper();
        try {
            return mapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {
            throw new RuntimeException(e);
        }
    }
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/JSONWebKeySet.kt
package io.jans.chip.model

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper

class JSONWebKeySet {
    private var keys: MutableList<Map<String?, *>?> = mutableListOf()
    fun addKey(key: Map<String?, *>?) {
        keys.add(key)
    }
    fun toJsonString(): String? {
        val mapper = ObjectMapper()
        return try {
            mapper.writeValueAsString(this)
        } catch (e: JsonProcessingException) {
            throw RuntimeException(e)
        }
    }
}